nixos/modules/services/networking/6tunnel.nix

```
@@ -0,0 +1,112 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services._6tunnel;
+
+in {
+  options.services._6tunnel = {
+
+    tunnels = mkOption {
+      default = [];
+      description = "Which tunnels to create.";
+      type = with types; listOf (submodule {
+        options = {
+          port = mkOption {
+            description = "Local port to open.";
+            type = types.port;
+          };
+          remoteHost = mkOption {
+            description = "Remote host to tunnel to.";
+            type = types.str;
+          };
+          remotePort = mkOption {
+            default = null;
+            description = "Port on remote host. If null (default), it uses the the local port.";
+            type = with types; nullOr port;
+          };
+          restart = mkOption {
+            default = null;
+            description = ''
+              Whether to restart the tunnel periodically, which is useful for non-static IP addresses.
+              The format is described in SYSTEMD.TIME(7).
+
+              If null (default), the tunnel won't restart periodically.
+            '';
+            type = with types; nullOr str;
+          };
+        };
+      });
+      example = [
+        { port = 8080; remoteHost = "192.168.0.2"; remotePort = 80; }
+        { port = 443; remoteHost = "www.example.com"; }
+        { port = 8053; remoteHost = "dynamicip.example.com"; restart = "1h"; }
+      ];
+    };
+
+    openFirewall = mkOption {
+      default = true;
```

```suggestion
      default = false;
```
Almost no service has this on `true` by default and we shouldn't confuse the users.